### PR TITLE
Improved enum support, fixes on serde tests

### PR DESCRIPTION
**Pull request by kali**
zonyitoo/bson-rs#44

### CODE REVIEW

File: serde-tests/Cargo.toml [hunk 1]
```diff
@@ -6,17 +6,17 @@ build = "build.rs"
...
 
 [dependencies]
-serde = "0.7"
+serde = "0.8"
 bson = { path = "..", features = ["serde"] }
 
 [build-dependencies]
-syntex = "^0.29.0"
-serde_codegen = "0.7"
+syntex = "*"
```
Comment (zonyitoo) [inline serde-tests/Cargo.toml]: I think crates.io will reject because of this `*`

Comment (kali) [inline serde-tests/Cargo.toml]: Mmm... this is in the Cargo.toml from serde-tests, which is not meant to be deployed to crates, but I can try and fix the version anyway.

**Comment by kali** [comment]
thanks :)

**Comment by zonyitoo** [comment]
Please test it in your project. I will publish it to crates.io lately.

**Comment by kali** [comment]
no problem, i can work without a crates.io version for a while.

**Pull request merged**